lightning/src/ln/interactivetxs.rs

```diff
@@ -430,6 +418,119 @@ impl InteractiveTxSigningSession {
+
+		write_tlv_fields!(writer, {
+			(0, self.unsigned_tx.holder_is_initiator, required),
+			(2, self.unsigned_tx.holder_sends_tx_signatures_first, required),
+			(4, self.has_received_commitment_signed, required),
+			(6, input_serial_ids, required),
+			(8, output_serial_ids, required),
+			(10, self.holder_tx_signatures, required),
+		});
+		Ok(())
+	}
+}
+
+impl Readable for InteractiveTxSigningSession {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let mut holder_is_initiator = false;
+		let mut holder_sends_tx_signatures_first = false;
+		let mut has_received_commitment_signed = false;
+		let mut input_serial_ids: Vec<SerialId> = vec![];
+		let mut output_serial_ids: Vec<SerialId> = vec![];
+		let mut holder_tx_signatures: Option<TxSignatures> = None;
+
+		read_tlv_fields!(reader, {
+			(0, holder_is_initiator, required),
+			(2, holder_sends_tx_signatures_first, required),
+			(4, has_received_commitment_signed, required),
+			(6, input_serial_ids, required),
+			(8, output_serial_ids, required),
+			(10, holder_tx_signatures, required),
+		});
+
+		// We use an empty dummy `TxOut` for all `TxOut`s as they will be populated with their real
+		// values from `funding_transaction` after reading.
+		let dummy_txout = TxOut { value: Amount::default(), script_pubkey: ScriptBuf::new() };
+
+		Ok(InteractiveTxSigningSession {
+			holder_sends_tx_signatures_first,
+			has_received_commitment_signed,
+			holder_tx_signatures,
+			unsigned_tx: ConstructedTransaction {
+				holder_is_initiator,
+				holder_sends_tx_signatures_first,
+				inputs: input_serial_ids
+					.into_iter()
+					.map(|serial_id| {
+						if is_serial_id_valid_for_counterparty(holder_is_initiator, serial_id) {
+							InteractiveTxInput::Remote(LocalOrRemoteInput {
+								serial_id,
+								input: TxIn::default(),
+								prev_output: dummy_txout.clone(),
+							})
+						} else {
+							InteractiveTxInput::Local(
+								// We use a dummy TxOut that will be populated from the funding_transaction after read.
+								LocalOrRemoteInput {
+									serial_id,
+									input: TxIn::default(),
+									prev_output: dummy_txout.clone(),
+								},
+							)
+						}
+					})
+					.collect(),
+				outputs: output_serial_ids
+					.into_iter()
+					.map(|serial_id| {
+						let added_by = if is_serial_id_valid_for_counterparty(
+							holder_is_initiator,
+							serial_id,
+						) {
+							AddingRole::Remote
+						} else {
+							AddingRole::Local
+						};
+						InteractiveTxOutput {
+							serial_id,
+							added_by,
+							// The ownership of the output is irrelevant at this point as the read would only happen
+							// at `ChannelState::InteractiveSigning`, so we just use `Single` here.
+							output: OutputOwned::Single(dummy_txout.clone()),
+						}
+					})
+					.collect(),
+				local_inputs_value_satoshis: 0,
+				local_outputs_value_satoshis: 0,
+				remote_inputs_value_satoshis: 0,
+				remote_outputs_value_satoshis: 0,
```

Comment:
Why are these zero? Should they be filled in from the funding transaction?

Reply:
Fixed as we now persist `InteractiveTxSigningSession` fully.